src/fidesops/api/v1/endpoints/policy_endpoints.py

```diff
@@ -155,7 +156,7 @@ def create_or_update_rules(
         verify_oauth_client,
         scopes=[scopes.RULE_CREATE_OR_UPDATE],
     ),
-    policy_key: str,
+    policy_key: FidesKey,
     db: Session = Depends(deps.get_db),
     input_data: conlist(schemas.RuleCreate, max_items=50) = Body(...),  # type: ignore
```

Comment:
`RuleCreate` schema contains a `FidesKey`, so this api endpoint can throw a 422 immediately on bulk create/updates instead of gracefully adding failed policies to a list and returning `succeeded` and `failed` policies (as is the current pattern in this endpoint). @pattisdr any thoughts on whether this is OK?

Reply:
This isn't introducing new behavior, this is the current pattern on our endpoints, even the bulk ones, because field-level validation happens before the code in the view runs (where we're creating the succeeded/failed responses).

For example, if you tried to create a Rule whose name was a `list` you'd also get a 422 immediately.  

In my opinion it's okay to fail early here, to save us from looping through potentially many resources that aren't formatted correctly.

Reply:
Got it, this makes total sense, thanks for clarifying :)